src/analysis/function_parameters.rs

```diff
@@ -198,12 +198,36 @@ pub fn analyze(
         .filter_map(|p| p.array_length.map(|pos| (pos, p)))
         .collect();
 
+    let mut to_remove = Vec::new();
+    let mut correction_instance = 0;
+
+    for par in function_parameters.iter() {
+        if par.scope.is_none() {
+            continue;
+        }
+        if let Some(index) = par.closure {
+            to_remove.push(index);
```

Comment:
You could take a look at how this is handled for `array` attributes. See e.g. `find_index_to_ignore` in `src/analysis/functions.rs`. Not sure if that improves anything though

Reply:
Also if `scope = notified` you would additionally take the `GDestroyNotify` at `index+1` and ignore that.

That seems to be handled kind of in `src/analysis/functions.rs:analyze_callbacks()`, so maybe that should go both into the same place?

Reply:
And there's e.g. `Trampoline::user_data_index` and `destroy_index`, so it seems like there kind of is infrastructure for this stuff already.

Reply:
Aaaand in this file there is `async_param_to_remove()`, which is completely wrong (it matches the parameter name too).

Reply:
Which is why I removed this function. ;)

Reply:
Right, forgot about that sorry :) The other things still apply though

Reply:
Absolutely, but first I'd like to understand why these changes still don't work. Then I'll clean up.